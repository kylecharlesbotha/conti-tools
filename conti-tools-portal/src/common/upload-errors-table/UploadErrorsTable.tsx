import {
  Box,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Theme,
  Typography
} from '@mui/material';
import { isEmpty } from 'lodash';

interface IUploadErrorsTable {
  errors?: Record<string, string[]>;
}

export const UploadErrorsTable = ({ errors }: IUploadErrorsTable) => {
  if (!errors || isEmpty(errors)) {
    return null;
  }

  return (
    <Box mt={2} sx={{ maxHeight: '250px', overflowY: 'auto' }}>
      <Typography
        variant="subtitle2"
        sx={{ color: (theme: Theme) => theme.palette.error.main }}
      >
        The following errors were found in the file:
      </Typography>
      <Table size="small">
        <TableHead>
          <TableRow>
            <TableCell>Line No / Field</TableCell>
            <TableCell>Error</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {Object.entries(errors).map(([key, value], errorIndex) => {
            return value.map((errorText, messageIndex) => (
              <TableRow key={`${errorIndex}-${messageIndex}`}>
                <TableCell>{key}</TableCell>
                <TableCell>{errorText}</TableCell>
              </TableRow>
            ));
          })}
        </TableBody>
      </Table>
    </Box>
  );
};
