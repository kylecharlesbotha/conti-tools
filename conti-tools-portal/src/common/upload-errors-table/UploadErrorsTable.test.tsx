import { render } from '@testing-library/react';
import { UploadErrorsTable } from './UploadErrorsTable';

describe('Testing rendering of the UploadErrorsTable component', () => {
  test('Without an errors param supplied it should render null', () => {
    const { baseElement } = render(<UploadErrorsTable />);
    expect(baseElement).toMatchSnapshot();
  });

  test('Without an empty errors array supplied it should render null', () => {
    const { baseElement } = render(<UploadErrorsTable errors={{}} />);
    expect(baseElement).toMatchSnapshot();
  });

  test('With some errors provided it should render the errors', () => {
    const errors = {
      input1: ['First Error', 'Second Error'],
      input2: ['First Error', 'Third Error']
    };

    const { baseElement } = render(<UploadErrorsTable errors={errors} />);
    expect(baseElement).toMatchSnapshot();
  });
});
