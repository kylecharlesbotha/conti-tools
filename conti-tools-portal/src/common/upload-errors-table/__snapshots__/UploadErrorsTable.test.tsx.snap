// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing rendering of the UploadErrorsTable component With some errors provided it should render the errors 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-dic6en"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-3l81ym-MuiTypography-root"
      >
        The following errors were found in the file:
      </h6>
      <table
        class="MuiTable-root css-rqglhn-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              Line No / Field
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              Error
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              input1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              First Error
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              input1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              Second Error
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              input2
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              First Error
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              input2
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              Third Error
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;

exports[`Testing rendering of the UploadErrorsTable component Without an empty errors array supplied it should render null 1`] = `
<body>
  <div />
</body>
`;

exports[`Testing rendering of the UploadErrorsTable component Without an errors param supplied it should render null 1`] = `
<body>
  <div />
</body>
`;
